src/rust/cryptography-x509-verification/src/policy/extension.rs

```diff
@@ -205,9 +205,10 @@ impl<B: CryptoOps> ExtensionValidator<B> {
             // Extension MUST NOT be present and isn't; OK.
             (ExtensionValidator::NotPresent, None) => Ok(()),
             // Extension MUST NOT be present but is; NOT OK.
-            (ExtensionValidator::NotPresent, Some(_)) => Err(ValidationError::Other(
-                "Certificate contains prohibited extension".to_string(),
-            )),
+            (ExtensionValidator::NotPresent, Some(extn)) => Err(ValidationError::Other(format!(
```

Comment:
IDK if this will make the PR way more complicated, but I think we shoudl do `ValidationError::ExtensionError` or something, and it can have the OID as structured data, and thus we can defer the string formatting until we're ready to show an error to the user.

WDYT?

Reply:
That wouldn't be too bad IMO -- I can either do that as part of this PR, or defer it to another as part of fixing https://github.com/pyca/cryptography/pull/11162#issuecomment-2190213329 as well, your preference :slightly_smiling_face:

Reply:
Let's do it here/now.